app/lib/hyrax/hirmeos/metrics_tracker.rb

```diff
@@ -6,10 +6,24 @@ def client
     @client ||= Hyrax::Hirmeos::Client.new(username, password, metrics_base_url, translation_base_url, secret)
   end
 
-  def submit_to_hirmeos(resource_id, resource_json)
-    response = client.get_resource(resource_id)
+  def submit_work_to_hirmeos(work_id, work_json)
+    response = client.get_resource(work_id)
     return if response.success?
-    client.post_resource(resource_json)
+    client.post_resource(work_json)
+  end
+
+  def submit_file_set_to_hirmeos(file_set_id, file_set_json)
+    response = client.get_resource(file_set_id)
+    json_response = JSON.parse(response.body)
+    return if file_already_registered?(json_response)
+    client.post_resource(file_set_json)
+  end
+
+  def file_already_registered?(json_response)
+    json_response["data"].each do |resource|
+      return true if resource['work']['type'] == "repository-file"
+    end
+    false
```

Comment:
This seems a little clunky. 

There are iterator methods like `any?` that do this for you. You would have to test it out and make sure your specs still pass, but something like this might work and be a little clearer: 
```
json_response["data"].any? { |resource| resource.dig("work", "type') == "repository-file" }
```

You're also using mixed hash access through out, `dig("data", 0, "URI")` in places `resource['work']['type']` in others. Try and use `dig` when you are digging into mutliple levels of an array.

Reply:
You're also using `'` and `"` in the same line. This project isn't Hyku Addons so we aren't trying to stick to just `"`, but you should try to be consistent anyway.

Reply:
Ill give it a go, thanks for the suggestion